src/pages/utilities/UnrecognizedUtilityTabs/LinkFilesTab.tsx

```diff
@@ -232,33 +242,108 @@ function LinkFilesTab() {
     return manualLinkFileRows;
   }, [selectedRows]);
 
-  const makeLinks = async () => {
-    await Promise.all(map(groupedLinksMap, async (fileIds, episodeID) => {
-      const payload: FileLinkApiType = {
-        episodeID: toNumber(episodeID),
-        fileIDs: [],
-      };
-      payload.episodeID = toNumber(episodeID);
-      forEach(fileIds, (link) => {
-        if (link.FileID === 0) {
-          return;
+  useEffect(() => {
+    if (!isLinking || isLinkingRunning || getAnidbSeriesQuery.isLoading || !(getAnidbSeriesQuery.isSuccess && getAnidbSeriesQuery.data.ShokoID)) {
+      return;
+    }
+
+    const seriesID = getAnidbSeriesQuery.data.ShokoID;
+    const linkFiles = async () => {
+      if (isLinkingRunning) return;
```

Comment:
This is duplicate of top condition

Reply:
Some explanation on what feature is being added here would be nice.

Reply:
> Some explanation on what feature is being added here would be nice.

We're making sure the Shoko series is only created when the 'Save' button is pressed, and not before that.